src/js/rise-cache.js

```diff
@@ -49,51 +49,53 @@ RiseVision.Common.RiseCache = (function () {
     }
 
     function fileRequest(isCacheRunning) {
-      var url, str, separator;
+      var xhr = new XMLHttpRequest(),
+        url, str, separator, request;
 
       if (isCacheRunning) {
         // configure url with cachebuster or not
         url = (nocachebuster) ? BASE_CACHE_URL + "?url=" + encodeURIComponent(fileUrl) :
           BASE_CACHE_URL + "cb=" + new Date().getTime() + "?url=" + encodeURIComponent(fileUrl);
+
+        // custom request object to provide in response
+        request = {
+          xhr: xhr,
+          url: url
+        };
+
+        xhr.open("GET", url, true);
+
+        xhr.addEventListener('loadend', function () {
+          var status = xhr.status || 0;
+
+          if (status === 0 || (status >= 200 && status < 300)) {
```

Comment:
@stulees @donnapep As I understood If there is a cross domain issue the status will be 0 so we have to consider it. cheers